cli/config/addressbook.go

```diff
@@ -0,0 +1,158 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/oasisprotocol/oasis-sdk/cli/addressbook"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/config"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
+)
+
+// AddressBook contains the configuration of the address book.
+type AddressBook struct {
+	// All is a map of all configured address entries in the address book.
+	All map[string]*AddressBookEntry `mapstructure:",remain"`
+}
+
+// Validate performs config validation.
+func (w *AddressBook) Validate() error {
+	// Make sure all entries are valid.
+	for name, acc := range w.All {
+		if err := config.ValidateIdentifier(name); err != nil {
+			return fmt.Errorf("malformed address name '%s': %w", name, err)
+		}
+
+		if err := acc.Validate(); err != nil {
+			return fmt.Errorf("address '%s': %w", name, err)
+		}
+	}
+
+	return nil
+}
+
+// Load loads the given address.
+func (w *AddressBook) Load(name string) (*addressbook.Address, error) {
+	cfg, exists := w.All[name]
+	if !exists {
+		return nil, fmt.Errorf("address account '%s' does not exist in the address book", name)
+	}
+
+	if err := config.ValidateIdentifier(name); err != nil {
+		return nil, fmt.Errorf("malformed account name '%s': %w", name, err)
+	}
+
+	af := addressbook.AddressFactory{}
+	acc, err := af.Load(name, cfg.Config, cfg.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make sure the address matches what we have in the config.
+	if expected, actual := cfg.GetAddress(), acc.Address(); !actual.Equal(expected) {
+		return nil, fmt.Errorf("address mismatch after loading account (expected: %s got: %s)",
+			expected,
+			actual,
+		)
+	}
+
+	return acc, nil
+}
+
+// Remove removes the given account.
+func (w *AddressBook) Remove(name string) error {
+	if _, exists := w.All[name]; !exists {
+		return fmt.Errorf("account '%s' does not exist in the address book", name)
+	}
+
+	if err := config.ValidateIdentifier(name); err != nil {
+		return fmt.Errorf("malformed account name '%s': %w", name, err)
+	}
+
+	delete(w.All, name)
+
+	return nil
+}
+
+// Rename renames an existing account.
```

Comment:
```suggestion
// Rename renames an existing address.
```
Also several cases below.